From 9fa393a287ae46785a0789086d4a3d3eda4d5059 Mon Sep 17 00:00:00 2001
From: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
Date: Fri, 3 Sep 2010 00:19:00 -0700
Subject: [PATCH 10/26] xen/evtchn: correction, pirq hypercall does not unmask

Unless one uses PHYSDEVOP_pirq_eoi_gmfn, which needs to be
investigated.

Signed-off-by: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
(cherry picked from commit cb23e8d58ca35b6f9e10e1ea5682bd61f2442ebd)

Signed-off-by: Ian Campbell <ijc@hellion.org.uk>
---
 drivers/xen/events.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/xen/events.c b/drivers/xen/events.c
index 045313c..220485c 100644
--- a/drivers/xen/events.c
+++ b/drivers/xen/events.c
@@ -437,11 +437,11 @@ static void pirq_eoi(int irq)
 	struct irq_info *info = info_for_irq(irq);
 	struct physdev_eoi eoi = { .irq = info->u.pirq.gsi };
 
-	if (unlikely(pirq_needs_eoi(irq))) {
+	if (pirq_needs_eoi(irq)) {
 		int rc = HYPERVISOR_physdev_op(PHYSDEVOP_eoi, &eoi);
 		WARN_ON(rc);
-	} else
-		unmask_irq(irq);
+	}
+	unmask_irq(irq);
 }
 
 static void pirq_query_unmask(int irq)
-- 
1.7.5.4

